server/src/main/java/org/elasticsearch/search/aggregations/timeseries/TimeSeriesAggregator.java

```diff
@@ -98,20 +90,22 @@ protected void doClose() {
 
     @Override
     protected LeafBucketCollector getLeafCollector(LeafReaderContext context, LeafBucketCollector sub) throws IOException {
-        final SortedBinaryDocValues tsids = tsidFieldData.load(context).getBytesValues();
+        // TODO: remove this method in a follow up PR
+        throw new UnsupportedOperationException("Shouldn't be here");
+    }
+
+    protected LeafBucketCollector getLeafCollector(LeafReaderContext context, LeafBucketCollector sub, AggregationExecutionContext aggCtx)
+        throws IOException {
         return new LeafBucketCollectorBase(sub, null) {
 
             @Override
             public void collect(int doc, long bucket) throws IOException {
-                if (tsids.advanceExact(doc)) {
-                    BytesRef newTsid = tsids.nextValue();
-                    long bucketOrdinal = bucketOrds.add(bucket, newTsid);
-                    if (bucketOrdinal < 0) { // already seen
-                        bucketOrdinal = -1 - bucketOrdinal;
-                        collectExistingBucket(sub, doc, bucketOrdinal);
-                    } else {
-                        collectBucket(sub, doc, bucketOrdinal);
-                    }
+                long bucketOrdinal = bucketOrds.add(bucket, aggCtx.getTsid());
```

Comment:
Should it add an assert to check aggCtx is not null, or check aggCtx null?

Reply:
We have a check much earlier that this aggregator runs in the correct environment to prevent NPE here. I don't really see the difference between failing it with NPE vs assert.